public/tutorials/using-infura-custom-provider.md

```diff
@@ -1,136 +1,152 @@
+<p class="alert alert-info">
+<strong>Note</strong>: If you are on Windows and get an `MSBUILD` error, you may need to install the Windows build tools. In a terminal with Administrator rights, run `npm install -g windows-build-tools` and then try installation again.
+</p>
 
-### Editing Your Project's Configuration
+## Register with Infura
 
-We can use your project's `truffle.js` configuration file to tell Truffle how to sign our transactions. To start, we should first require all of our dependencies used for signing transactions:
+Before you can use Infura, you need to register for an Infura Access Token.
 
-```javascript
-var bip39 = require("bip39");
-var hdkey = require('ethereumjs-wallet/hdkey');
-var ProviderEngine = require("web3-provider-engine");
-var WalletSubprovider = require('web3-provider-engine/subproviders/wallet.js');
-var Web3Subprovider = require("web3-provider-engine/subproviders/web3.js");
-var Web3 = require("web3");
-const FilterSubprovider = require('web3-provider-engine/subproviders/filters.js');
-```
+You can do this at https://infura.io/register.html. Fill out the form, and you will receive your token. Your information will be sent to your email and displayed on the screen. Make sure you save this token and keep it private!
 
-These dependencies are used for the following things:
 
-- `bip39`: Used to create a seed from hd wallet mnemonics (more on that later).
-- `hdkey`: Used to derive addresses from seeds using complex math.
-- `ProviderEngine`: The basic framework that will be used to wrangle all transactions that need to be signed.
-- `WalletSubprovider`: Part of the provider engine framework that will handle wallet signing.
-- `Web3Subprovider`: Part of the provider engine framework that handles everything *other than* transaction signing.
-- `Web3`: What we used to communicate with the Ethereum network. Here, we're using it solely to create a provider.
+## Configure your Truffle project
 
-After setting up our dependencies, we'll then need to create an hd wallet from a mnemonic. A mnemonic is a twelve word string that represents a secure random seed (warning: don't just create a mnemonic yourself; use the bip39 library to do this if you don't have one). From this seed we can create an unlimited number of Ethereum addresses and private keys. Because this is a seed, we can always use this mnemonic to find those addresses again.
+The next step is to edit your `truffle.js` file to use `HDWalletProvider` and provide all the necessary configuration for deploying to Ropsten.
 
-```javascript
-// Get our mnemonic and create an hdwallet
-var mnemonic = "couch solve unique spirit wine fine occur rhythm foot feature glory away";
-var hdwallet = hdkey.fromMasterSeed(bip39.mnemonicToSeed(mnemonic));
-```
+1. First, define the provider in the configuration. Add this line at the top of your `truffle.js` file:
```

Comment:
I would say `define the HDWalletProvider object within your configuration file` as `define the provider` could be confused with setting the `provider` key in your desired network (a step done later)